plugins/modules/aws_s3.py

```diff
@@ -700,7 +814,9 @@ def main():
         content=dict(),
         content_base64=dict(),
         ignore_nonexistent_bucket=dict(default=False, type='bool'),
-        encryption_kms_key_id=dict()
+        encryption_kms_key_id=dict(),
+        tags=dict(type='dict'),
+        purge_tags=dict(type='bool', default=True),
```

Comment:
Why was this deemed a good idea?
Now, everything that touches any asset that uses tags MUST now add a `purge_tags: False` just to leave things as they are??

Use case:
```
- community.aws.ec2_asg:
    name: "{{ item.auto_scaling_group_name }}"
    min_size: "{{ count }}"
    max_size: "{{ count }}"
    desired_capacity: "{{ count }}"
    region: eu-central-1
```

Used to allow modifying an ASG min/max/desired size.

Now, does the above and **strips all tags**, how is that the "principle of least surprising behavior"?

This requires adding the `purge_tags: False` **everywhere** where tags could be impacted.

```
- community.aws.ec2_asg:
    name: "{{ item.auto_scaling_group_name }}"
    min_size: "{{ count }}"
    max_size: "{{ count }}"
    desired_capacity: "{{ count }}"
    purge_tags: False
    region: eu-central-1
```

This is the ONLY case where one has to specify a non-default option to "leave things alone".